docs/development/dependencies.md

```diff
@@ -0,0 +1,138 @@
+# Managing dependencies with Poetry
+
+This is a quick cheat sheet for developers on how to use [`poetry`](https://python-poetry.org/).
```

Comment:
It might be worth some explanatory/additional background here. E.g.
- broad dependencies versus the locked environment.
- the motivation for locking dependencies at all. 
- a link to the issue where I recommended poetry in the first place
- comparison to other ecosystems (npm/cargo. Ruby? Go?)

Reply:
TBH I think the only item from this list I might add would be the link to the issue where you recommend poetry-the rest feel a little more like blog-post material vs docs material and may risk overwhelming the user. That being said, this doc is already great and I think people will find it quite helpful.

Reply:
I've added a "background section" which address some of this. It might be overwhelming but I think it's useful to have examples from the project file and the lockfile. I didn't mention anything in the fourth bullet, and only added a link for the third. Hopefully this is a reasonable compromise? /shrug